world-chain-builder/crates/world/payload/src/builder.rs

```diff
@@ -616,11 +615,23 @@ where
             }
 
             // If the transaction is verified, check if it can be added within the verified gas limit
-            if pooled_tx.valid_pbh()
-                && info.cumulative_gas_used + tx.gas_limit() > verified_gas_limit
-            {
-                best_txs.mark_invalid(tx.signer(), tx.nonce());
-                continue;
+            if pooled_tx.valid_pbh() {
+                if info.cumulative_gas_used + tx.gas_limit() > verified_gas_limit {
+                    best_txs.mark_invalid(tx.signer(), tx.nonce());
+                    continue;
+                }
+
+                let Some(payloads) = pooled_tx.payload() else {
+                    unimplemented!()
+                };
```

Comment:
It looks like there will never be a case where `pooled_tx.valid_pbh()` will be true but `pooled_tx.payload()` will be `None`.

Since the implementation of `valid_pbh()` is:
```rust
  fn valid_pbh(&self) -> bool {
        self.payload.is_some()
    }

```

I think we can consolidate this to `if let Some(payload) = pooled_tx.pbh_payload()` to check if the tx is PBH.